test/hook/context/context_combiner.go

```diff
@@ -0,0 +1,90 @@
+package context
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/flant/shell-operator/pkg/hook/binding_context"
+	"github.com/flant/shell-operator/pkg/hook/controller"
+	"github.com/flant/shell-operator/pkg/hook/task_metadata"
+	"github.com/flant/shell-operator/pkg/hook/types"
+	shell_operator "github.com/flant/shell-operator/pkg/shell-operator"
+	"github.com/flant/shell-operator/pkg/task"
+	"github.com/flant/shell-operator/pkg/task/queue"
+)
+
+const TestTaskType task.TaskType = "TestTask"
+
+type ContextCombiner struct {
+	op *shell_operator.ShellOperator
+	q  *queue.TaskQueue
+}
+
+func NewContextCombiner() *ContextCombiner {
+	op := &shell_operator.ShellOperator{}
+	op.TaskQueues = queue.NewTaskQueueSet()
+	op.TaskQueues.WithContext(context.Background())
+	op.TaskQueues.NewNamedQueue("main", nil)
+
+	return &ContextCombiner{
+		op: op,
+		q:  op.TaskQueues.GetByName("main"),
+	}
+}
+
+func (c *ContextCombiner) AddBindingContext(bindingType types.BindingType, info controller.BindingExecutionInfo) {
+	t := task.NewTask(TestTaskType)
+	t.WithMetadata(task_metadata.HookMetadata{
+		Binding:        info.Binding,
+		Group:          info.Group,
+		BindingType:    bindingType,
+		BindingContext: info.BindingContext,
+		AllowFailure:   info.AllowFailure,
+	})
+	t.WithQueueName("main")
+	c.q.AddLast(t)
+}
+
+// CombinedContext returns a combined context or a binding context
+// from the first task - as a Shell-operator will do.
+func (c *ContextCombiner) Combined() []binding_context.BindingContext {
+	firstTask := c.q.GetFirst()
+	if firstTask == nil {
+		return []binding_context.BindingContext{}
+	}
+	taskMeta := firstTask.GetMetadata()
+
+	bc := taskMeta.(task_metadata.BindingContextAccessor).GetBindingContext()
```

Comment:
Why do we need to cast a type here?

Reply:
Metadata is interface{} to have different HookMetadata in shell-operator and in addon-operator.